src/lib/extract-acl-context.ts

```diff
@@ -32,15 +28,37 @@ const getPermissions = (config: RuntimeConfig, request: FastifyRequest): string[
   return parsedProperties.permissions ?? []
 }
 
-export const extractAclContext = (
+const cleanHeaders = (headers: Record<string, string | string[] | undefined>): Record<string, string | string[] | undefined> => {
+  delete headers.cookie
```

Comment:
this `delete` operation is in-place.. maybe something else is holding a ref to this object. defs fastify request handler does. Maybe spread-clone the object